internal/datastore/common/migrations_test.go

```diff
@@ -0,0 +1,61 @@
+package common
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/require"
+
+	"github.com/authzed/spicedb/pkg/datastore"
+)
+
+func TestMigrationReadyState(t *testing.T) {
+	tests := []struct {
+		name                        string
+		additionalAllowedMigrations []string
+		headMigration               string
+		version                     string
+		want                        datastore.ReadyState
+	}{
+		{
+			name:          "matches head migration",
+			headMigration: "initial",
+			version:       "initial",
+			want:          datastore.ReadyState{IsReady: true},
+		},
+		{
+			name:                        "matches additional migration",
+			headMigration:               "initial",
+			additionalAllowedMigrations: []string{"additional"},
+			version:                     "additional",
+			want:                        datastore.ReadyState{IsReady: true},
+		},
+		{
+			name:          "matches head migration",
+			headMigration: "initial",
+			version:       "initial",
+			want:          datastore.ReadyState{IsReady: true},
+		},
```

Comment:
Dupe of top test?

Reply:
nice catch! this was supposed to be a case matching the head revision but with additional ones allowed.